src/services/plugin.service.ts

```diff
@@ -12,6 +13,24 @@ export class PluginService {
     if (this.plugins[pluginName]) {
       throw new Error(`${pluginName} already exists as a plugin.`);
     }
-    return (this.plugins[pluginName] = new PluginLoader(pluginName, args) as IPlugin);
+
+    const reference = (this.plugins[pluginName] = new PluginLoader(pluginName, args) as IPlugin);
+
+    this.registerCommands(pluginName);
+
+    return reference;
+  }
+
+  registerCommands(pluginName: string, args?: any): void {
+
+    const commands = this.plugins[pluginName].pluginCommands;
+
+    if (!commands.includes(pluginName)) {
+      this.commands[pluginName] = pluginName;
+    }
+
+    commands.forEach((command: string, index: number) => {
+      this.commands[command] === undefined ? this.commands[command] = pluginName : this.plugins[pluginName].pluginCommands.splice(index, 1);
```

Comment:
sketchy. Instead, pass over an already-used alias and log a warning.

```
if (this.commands[command]) {
  // log warning
} else {
  this.commands[command] = pluginName;
}

```

Reply:
I'll look at the other warning logs and implement it along with the other noted improvements.